lua/rustaceanvim/dap.lua

```diff
@@ -270,41 +264,55 @@ function M.start(args)
         return
       end
 
-      -- create debug configuration
-      local dap_config = {
-        name = 'Rust tools debug',
-        type = 'rt_lldb',
-        request = 'launch',
-        program = executables[1],
-        args = args.executableArgs or {},
-        cwd = args.workspaceRoot,
-        stopOnEntry = false,
-
-        -- if you change `runInTerminal` to true, you might need to change the yama/ptrace_scope setting:
-        --
-        --    echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
-        --
-        -- Otherwise you might get the following error:
-        --
-        --    Error on launch: Failed to attach to the target process
-        --
-        -- But you should be aware of the implications:
-        -- https://www.kernel.org/doc/html/latest/admin-guide/LSM/Yama.html
-        runInTerminal = false,
-      }
-      local final_config = next(init_commands) ~= nil
-          and vim.tbl_deep_extend('force', dap_config, { initCommands = init_commands[args.workspaceRoot] })
-        or dap_config
+      -- If the `lldb` adapter is not defined elsewhere, use the adapter
+      -- defined in `config.dap.adapter`
+      if dap.adapters.lldb == nil and adapter ~= false then
```

Comment:
This can be moved into the `dap.adapter` configuration for better transparency/control.

It won't do any harm being here, but it will make things 'just work' so it could cause some confusion for users who are setting up their own `lldb` adapter.

Configuration of the `lldb` adapter happens here, instead of in the `dap.adapter` configuration function as an `lldb` adapter can be added at any point in the users configuration.

However,

Reply:
It looks like some of your text got cut off?

> Configuration of the lldb adapter happens here, instead of in the dap.adapter configuration function as an lldb adapter can be added at any point in the users configuration.

To solve this, we can call `types.evalueate(config.dap.adapter)` within the `dap.start()` function, and pass it down to wherever it's needed as a function parameter.

I can implement that after this is merged though :smile:

Reply:
Whoops, yes it did.

The part that got cutoff was the bit explaining why I don't think that matters.
If the user config comes along later, it will overwrite what `rustacean` has done; which is the intended outcome.

Either way suits me :smile: